### PR TITLE
Default to project level lock file

### PR DESCRIPTION
**Pull request by ljacomet**
The lockfile per project is now the default and only format written when
creating or updating lock files.
The previous format remains supported for loading lock state which
enables to migrate to the new format configuration per configuration.

### CODE REVIEW

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

**Comment by ljacomet** [comment]
@bot-gradle test RFM

File: subprojects/core/src/main/java/org/gradle/api/internal/FeaturePreviews.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
     public enum Feature {
         GRADLE_METADATA(false),
         GROOVY_COMPILATION_AVOIDANCE(true),
-        ONE_LOCKFILE_PER_PROJECT(true),
+        ONE_LOCKFILE_PER_PROJECT(false),
```
Comment (bmuskalla) [inline subprojects/core/src/main/java/org/gradle/api/internal/FeaturePreviews.java:31]: Is this now still needed?

**Review by bmuskalla — APPROVED** [review]
LGTM, small question inline as keeping the feature flag while removing all the tests is a bit contradicting

File: subprojects/core/src/main/java/org/gradle/api/internal/FeaturePreviews.java (hunk 1 above)
Comment (ljacomet) [reply subprojects/core/src/main/java/org/gradle/api/internal/FeaturePreviews.java:31]: Yes, it now prints a deprecation message when used and we remove it in 8.0. Like `GRADLE_METADATA` which has to be removed for 7.0.

**Pull request merged**